File: ServerSubscriptionManager/Models/SubscriptionPeriod.cs
namespace ServerSubscriptionManager.Models
{
    public class SubscriptionPeriod
    {
        public long Id { get; set; }
        public required DateOnly StartDate { get; set; }
        public required DateOnly EndDate { get; set; }
        public required decimal ServerCost { get; set; }
        public int UserCount { get; set; } = 0;
        public decimal UserCost => UserCount > 0 ? ServerCost / UserCount : ServerCost;
        public decimal NextUserCost => (UserCount + 1) > 0 ? ServerCost / (UserCount + 1) : ServerCost;
        public ICollection<Invoice> Invoices { get; set; } = [];
    }
}

File: web/src/app/models/subscription-period.ts
import { Invoice } from "./invoice";

export interface SubscriptionPeriod {
    id: number;
    startDate: string;
    endDate: string;
    serverCost: number;
    userCount: number;
    userCost: number;
    nextUserCost: number;
    invoices: Invoice[];
}